x-pack/legacy/plugins/siem/public/pages/case/components/__mock__/form.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+export const mockFormHook = {
+  isSubmitted: false,
+  isSubmitting: false,
+  isValid: true,
+  submit: jest.fn(),
+  subscribe: jest.fn(),
+  setFieldValue: jest.fn(),
+  setFieldErrors: jest.fn(),
+  getFields: jest.fn(),
+  getFormData: jest.fn(),
+  getFieldDefaultValue: jest.fn(),
+  /* Returns a list of all errors in the form */
+  getErrors: jest.fn(),
+  reset: jest.fn(),
+  __options: {},
```

Comment:
Do we need to mock these internal functions (`__addField` etc)?

Reply:
yes, otherwise we get type errs `TypeError: form.__readFieldConfigFromSchema is not a function`